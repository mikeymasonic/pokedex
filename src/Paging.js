import React, { Component } from 'react';
export default class Paging extends Component {
    state = 
    { page: 1, 
      perPage: 20,
    }
    changePage(value) {
        this.setState({ page: this.state.page + value})
        const queryString = window.location.hash.slice(1);
        const pageParams = new URLSearchParams(queryString);
        pageParams.set('page', this.state.page + value);
        pageParams.set('perPage', this.state.perPage)
        const paramToSet = pageParams.toString();
        window.location.hash = paramToSet;
    }
    componentDidMount() {
        const pageParams = new URLSearchParams();
        pageParams.set('page', this.state.page);
        const paramToSet = pageParams.toString();
        window.location.hash = paramToSet;
        window.addEventListener('hashchange', () => {
            const queryString = window.location.hash.slice(1);
            const pageParams = new URLSearchParams(queryString);
            let pageToUse = this.state.page;
            const parsedPage = Number(pageParams.get("page"));
            if (isNaN(parsedPage)) {
              pageToUse = 1;
            } else {
              pageToUse = parsedPage;
            }       
            this.setState({ page: pageToUse });
        })
    }
    render() {
        if (!this.props.totalPage) {
            return <p className="paging">No results, try another search</p>;
          }

        return (
            <div className="paging">
                <button id="paging-button" onClick={() => this.changePage(-1)} disabled={this.state.page === 1}> LAST </button>
                <button id="paging-button" onClick={() => this.changePage(1)} disabled={this.state.page === this.props.totalPage}> NEXT </button>
                <h5>Page {this.state.page}/{this.props.totalPage}</h5>
            </div>
        )
    }
}