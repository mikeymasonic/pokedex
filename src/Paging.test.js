import React from 'react';
import Paging from './Header';
import renderer from "react-test-renderer";
test('renders Paging.js correctly', () => {
  const tree = renderer
  .create(<Paging />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});