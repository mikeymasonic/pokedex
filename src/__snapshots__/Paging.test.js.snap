// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Paging.js correctly 1`] = `
<div>
  <header>
    <a
      href="./"
    >
      <img
        alt="Pokemon"
        className="logo"
        src="https://i.ebayimg.com/images/g/8bMAAOSwxp9W-TVs/s-l400.jpg"
      />
    </a>
    <h1>
      Pokedex
    </h1>
    <div
      id="search"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          id="search"
          onChange={[Function]}
          value=""
        />
        <button
          id="submit"
        >
          Search
        </button>
      </form>
    </div>
  </header>
</div>
`;
